cli/azd/pkg/tools/azcli/azcli.go

```diff
@@ -295,16 +301,21 @@ type Filter struct {
 type NewAzCliArgs struct {
 	EnableDebug     bool
 	EnableTelemetry bool
-	HttpClient      httputil.HttpClient
 }
 
-func NewAzCli(credential azcore.TokenCredential, args NewAzCliArgs) AzCli {
+func NewAzCli(
+	subscriptionTenantResolver account.SubscriptionTenantResolver,
+	credentialProvider auth.MultiTenantCredentialProvider,
+	httpClient httputil.HttpClient,
+	args NewAzCliArgs,
+) AzCli {
 	return &azCli{
-		userAgent:       azdinternal.MakeUserAgentString(""),
-		enableDebug:     args.EnableDebug,
-		enableTelemetry: args.EnableTelemetry,
-		httpClient:      args.HttpClient,
-		credential:      credential,
+		credentialProvider:         credentialProvider,
+		enableDebug:                args.EnableDebug,
+		enableTelemetry:            args.EnableTelemetry,
+		httpClient:                 httpClient,
+		subscriptionTenantResolver: subscriptionTenantResolver,
```

Comment:
Should we create a new type struct to group the credentialProvider and the subscriptionTenantResolver?

That would help to expose something like `credentialFactory.subCredential(subscriptionId string)`  and the type can be use ouside the `azCli`.   It can be added to the container and injected to `azCli` and to any other component which might require the tokenCredential in the future.

For example, eventually, when we start breaking `azCli` apart, the new components can start consuming this type, instead of consuming `credentialProvider + subscriptionTenantResolver` and implementing the `subCredential()` everywhere.

Reply:
Thanks, I was thinking about this as well.  It was easier to bang out the direct version, so that's where I started, but I will introduce an abstraction in the `account` package that has an interface like:

```
type GetSubscriptionCredential interface {
  GetSubscriptionCredential(subId string) (azcore.TokenCredential, error)
}
```

And inject this interface instead of the two components.

I don't think I will merge `auth` and `account` at this point but that might be the right thing to do long term.